transforms-caching-experiment/remote-cache-experiment-transforms.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# REQUIRED: Having the key set in the environment is expected - note that remote build cache write is requiered. It can be set here or exported in the terminal before running the script.
+#export DEVELOCITY_ACCESS_KEY=$develocityUrl=<access-key>
+
+gradleVersion=$(./gradlew --version | grep "Gradle" | awk '{print $2}')
+
+major=$(echo "$gradleVersion" | cut -d. -f1)
+minor=$(echo "$gradleVersion" | cut -d. -f2)
+
+# Check if the version is lower than 8.9
+if [ "$major" -lt 8 ] || { [ "$major" -eq 8 ] && [ "$minor" -lt 9 ]; }; then
+  echo "Gradle version $gradleVersion is lower than 8.9"
+  exit 1
+else
+    echo "Gradle version detected: $gradleVersion"
+fi
+
+homeDir=build/HOME
+
+# Set 'task' to the first argument or 'help' if no arguments are provided
+tasks=${*:-help}
+
+# Initialize empty Gradle User Home with settings to run build
+echo "Initializing Gradle User Home directory at $homeDir"
+rm -rf $homeDir
+mkdir -p $homeDir
+mkdir -p $homeDir/caches/"$gradleVersion"/
+cp ~/.gradle/gradle.properties $homeDir
+cp -r ~/.gradle/caches/"$gradleVersion"/generated-gradle-jars $homeDir/caches/"$gradleVersion"/
+
+# Note: This is expecting that CCUD gradle plugin is applied
+export GRADLE_CACHE_REMOTE_PUSH=true
+export GRADLE_CACHE_REMOTE_PATH="cache/$USER-exp-non-task"
+#export GRADLE_CACHE_REMOTE_URL="<develocityUrl>/cache/$USER-exp-non-task"  # Needed if the HTTP cache connector is used
+
+echo "------------------------------------------------------------"
+echo "Priming build with task '$tasks' and HOME=$homeDir"
+echo "------------------------------------------------------------"
+set -x
+# shellcheck disable=SC2086
+./gradlew $tasks -g $homeDir -Dscan.tag.remote-cache-experiment-init --no-configuration-cache -Ddevelocity.deprecation.muteWarnings=true -Dscan.uploadInBackground=false -Dgradle.cache.local.enabled=false
+set +x
+
+runs='transforms'
+# runs='transforms transforms-selected' # Uncomment to test with selected transforms disabled
+
+for run in $runs
+do
+    # Set args based on cache
+    if [ "$run" == 'transforms' ]
+    then
+        disabledCacheArgs='-Dorg.gradle.internal.transform-caching-disabled'
+    elif [ "$run" == 'transforms-selected' ]
+    then
+        # Specify the transforms to disable. Example below:
+        disabledTransforms='org.jetbrains.kotlin.gradle.internal.transforms.BuildToolsApiClasspathEntrySnapshotTransform,org.jetbrains.kotlin.gradle.internal.transforms.ClasspathEntrySnapshotTransform'
+        disabledCacheArgs="-Dorg.gradle.internal.transform-caching-disabled=$disabledTransforms"
+    fi
+
+    for args in "-Dscan.tag.baseline-$run" "-Dscan.tag.disabled-cache-$run $disabledCacheArgs"
+    do
+        echo "------------------------------------------------------------"
+        echo "Test caches/*/transforms removal with $args"
+        echo "------------------------------------------------------------"
+        set -x
+        ./gradlew --stop
+        killall -9 java
```

Comment:
This was based on Daz's script implemented for testing whether adding the experimental flag used here would make sense. As such, I believe it's to ensure any running daemons aren't interefering with deleting transforms from Gradle user home below and to make sure any runs are actually clean.

Reply:
Okay, I think this would be pretty destructive if all Java process are killed. At the very least it should be called out in the README with an obvious note. Ideally, we can minimize the blast radius to killing only relevant processes and I still would include a caution call-out in the README. 

I seem to recall doing this in the past and it even killing IntelliJ. You may want to check. With the BVS you can invoke the scripts and let them run in the background while you continue to do other work. With this line, it might kill whatever you're working on.

Reply:
I tested running without the `killall`, which made the daemon to be reused and thus results were corrupted, as we got pretty much everything up-to-date. I did change it to `pkill` to kill only gradle daemons. Note that even with `killall` I didn't see any issues with idea, that continued to work just fine. 

- [killall execution](https://ge.solutions-team.gradle.com/scans?search.tags=killall&search.timeZoneId=Europe%2FLjubljana&search.usernames=gkojek)
- [no kill execution](https://ge.solutions-team.gradle.com/scans?search.tags=no-killall&search.timeZoneId=Europe%2FLjubljana&search.usernames=gkojek)
- [pkill execution](https://ge.solutions-team.gradle.com/scans?search.tags=pkill&search.timeZoneId=Europe%2FLjubljana&search.usernames=gkojek)

Reply:
Oh, thanks, I haven't even thought of that

Reply:
nvm, we are controlling the way we launch `gradlew` anyway

Reply:
> nvm, we are controlling the way we launch gradlew anyway

Yes, but the Daemon could be disabled in `gradle.properties`.

If you invoke the builds with `--no-daemon` do you need to kill anything since it will create a single use Daemon? Would that be a better option?

Reply:
Changed to `--no-daemon`. Works the same way from my testing. I removed the `pkill` and `./gradlew --stop` commands from the script, as well as the notice in the README.